gwswlib/hydx.py

```diff
@@ -0,0 +1,242 @@
+        {"Label": "Metselwerk (bepleisterd)", "Code": "MSW"},
+        {"Label": "Metselwerk (onbepleisterd)", "Code": "MSW"},
+        {"Label": "HDPE", "Code": "HPE"},
+        {"Label": "Polyester", "Code": "HPE"},
+        {"Label": "Polyetheen", "Code": "HPE"},
+        {"Label": "Polypropyleen", "Code": "HPE"},
+        {"Label": "Plaatijzer", "Code": "PIJ"},
+        {"Label": "Staal", "Code": "STL"},
+    ]
+
+    VORMPUTCOLL = [
+        {"Label": "Rechthoekig", "Code": "RHK"},
+        {"Label": "Rond", "Code": "RND"},
+    ]
+
+    TYPEKNOOPPUNTCOLL = [
+        {"Label": "Inspectieput", "Code": "INS"},
+        {"Label": "Infiltratieput", "Code": "ITP"},
+        {"Label": "Compartiment", "Code": "CMP"},
+        {"Label": "Uitlaat", "Code": "UIT"},
+    ]
+
+    STATUSOBJECTCOLL = [
+        {"Label": "In gebruik", "Code": "ACT"},
+        {"Label": "In ontwerp", "Code": "ONT"},
+    ]
+
+    AANNAMEHYDXCOLL = [
+        {"Label": "Expert judgement", "Code": "EXJ"},
+        {"Label": "Conform ontwerp", "Code": "ONT"},
+        {"Label": "Conform norm", "Code": "NRM"},
+    ]
+
+    FIELDS = [
+        {
+            "csvheader": "UNI_IDE",
+            "fieldname": "HydxIdentificatieKnooppuntOfVerbinding",
+            "unit": "-",
+            "type": ('string', 10),
+            "required": True,
+        },
+        {
+            "csvheader": "RST_IDE",
+            "fieldname": "HydxIdentificatieRioolstelsel",
+            "unit": "-",
+            "type": ('string', 30),
+            "required": False,
+        },
+        {
+            "csvheader": "PUT_IDE",
+            "fieldname": "HydxIdentificatieRioolput",
+            "unit": "-",
+            "type": ('string', 10),
+            "required": True,
+        },
+        {
+            "csvheader": "KNP_XCO",
+            "fieldname": "HydxX_coordinaat",
+            "unit": "m",
+            "type": ('float', 12.2),
+            "required": True,
+        },
+        {
+            "csvheader": "KNP_YCO",
+            "fieldname": "HydxY_coordinaat",
+            "unit": "m",
+            "type": ('float', 12.2),
+            "required": True,
+        },
+        {
+            "csvheader": "CMP_IDE",
+            "fieldname": "HydxIdentificatieCompartiment",
+            "unit": "-",
+            "type": ('string', 10),
+            "required": False,
+        },
+        {
+            "csvheader": "MVD_NIV",
+            "fieldname": "HydxNiveauMaaiveld",
+            "unit": "m",
+            "type": ('float', 8.2),
+            "required": True,
+        },
+        {
+            "csvheader": "MVD_SCH",
+            "fieldname": "HydxMaaiveldschematisering",
+            "unit": "-",
+            "type": ('string', 3),
+            "required": True,
+        },
+        {
+            "csvheader": "WOS_OPP",
+            "fieldname": "HydxOppervlakWaterOpStraat",
+            "unit": "m2",
+            "type": ('float', 8.2),
```

Comment:
Ik vraag me af of we iets moeten doen met die `8.2`-achtige lengte-zaken. Als we het in een CSVtje wegschrijven om het later te importeren, dan pikt die ander het wel op dat 'ie wil afronden bijvoorbeeld.

Met strings moeten we misschien een tikkie strenger zijn. Maar dan nog: is het echt nodig? Ik vraag me af of het implementatiedetails van hun database zijn of dat het *echte* beperkingen zijn.

Reply:
Heb verwijderd